File: app/src/main/java/com/sachtech/datingapp/ui/explore/fragment/ExploreFragment.java
package com.sachtech.datingapp.ui.explore.fragment;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;

import com.google.android.material.tabs.TabLayout;
import com.google.android.material.tabs.TabLayout.BaseOnTabSelectedListener;
import com.google.android.material.tabs.TabLayout.Tab;
import com.sachtech.datingapp.R;
import com.sachtech.datingapp.R.id;
import com.sachtech.datingapp.cometChat.CometChatMessaging;
import com.sachtech.datingapp.networking.FirebaseHelper;
import com.sachtech.datingapp.ui.explore.adapter.SearchAdapter;
import com.sachtech.datingapp.ui.explore.adapter.listener.CallTypeListener;
import com.sachtech.datingapp.ui.home.listener.OnFragmentChange;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.ArrayList;

import kotlin.jvm.internal.Intrinsics;

public final class ExploreFragment extends Fragment implements CallTypeListener {

   @Nullable
   private SearchAdapter searchAdapter;
   @Nullable
   private ArrayList arrayList;

   @Nullable
   private OnFragmentChange onFragmentChange;

   public void onAudioCallSelected(@NotNull String uid) {
      this.getCometMessage().initiateCall(uid, "audio");
   }

   public void onVideoCallSelected(@NotNull String uid) {
      this.getCometMessage().initiateCall(uid, "video");
   }

   @NotNull
   public final CometChatMessaging getCometMessage() {

      return new CometChatMessaging();
   }

   @Nullable
   public final SearchAdapter getSearchAdapter() {
      return this.searchAdapter;
   }

   public final void setSearchAdapter(@Nullable SearchAdapter var1) {
      this.searchAdapter = var1;
   }

   @Nullable
   public final ArrayList getArrayList() {
      return this.arrayList;
   }

   public final void setArrayList(@Nullable ArrayList var1) {
      this.arrayList = var1;
   }

   @NotNull
   public final FirebaseHelper getFirebaseHelper() {
      return new FirebaseHelper();
   }

   @Nullable
   public final OnFragmentChange getOnFragmentChange() {
      return this.onFragmentChange;
   }

   public final void setOnFragmentChange(@Nullable OnFragmentChange var1) {
      this.onFragmentChange = var1;
   }

   public void onAttach(@NotNull Context context) {
      super.onAttach(context);
      OnFragmentChange var10000 = this.onFragmentChange;
      if (var10000 != null) {
         var10000.selectedPos(2);
      }

   }

   @androidx.annotation.Nullable
   @Override
   public View onCreateView(@NonNull LayoutInflater inflater, @androidx.annotation.Nullable ViewGroup container, @androidx.annotation.Nullable Bundle savedInstanceState) {
      return LayoutInflater.from(getContext()).inflate(R.layout.fragment_search,container,false);
   }

   public void onViewCreated(@NotNull View view, @Nullable Bundle savedInstanceState) {
      super.onViewCreated(view, savedInstanceState);
      OnFragmentChange var10000 = this.onFragmentChange;
      if (var10000 != null) {
         var10000.selectedPos(2);
      }

      this.arrayList = new ArrayList();
     setOnClickListener(view);
      this.openFragment((Fragment) (new LikesFragment()));
      this.searchAdapter = new SearchAdapter(this.arrayList, (CallTypeListener) this);
   }

   private  void setOnClickListener(View view) {
      ((TabLayout) view.findViewById(id.tabLayout)).setOnTabSelectedListener((BaseOnTabSelectedListener) (new BaseOnTabSelectedListener() {
         public void onTabReselected(@Nullable Tab p0) {
         }

         public void onTabUnselected(@Nullable Tab p0) {
         }

         public void onTabSelected(@Nullable Tab p0) {
            if (p0 == null) {
               Intrinsics.throwNpe();
            }

            switch (p0.getPosition()) {
               case 0:
                  ExploreFragment.this.openFragment((Fragment) (new LikesFragment()));
                  break;
               case 1:
                  ExploreFragment.this.openFragment((Fragment) (new VisitorsFragment()));
                  break;
               case 2:
                  ExploreFragment.this.openFragment((Fragment) (new LikedByMeFragment()));
                  break;
               case 3:
                  ExploreFragment.this.openFragment((Fragment) (new NotInterestedUserFragment()));
                  break;
               case 4:
                  ExploreFragment.this.openFragment((Fragment) (new BlockUserFragment()));
            }

         }
      }));
   }

   private final void openFragment(Fragment kClass) {
      FragmentTransaction var10000 = this.getChildFragmentManager().beginTransaction();
      FragmentTransaction transaction = var10000;
      transaction.replace(id.exploreView, kClass).commit();
   }

}

File: app/src/main/java/com/sachtech/datingapp/ui/explore/fragment/ExploreFragment.kt
/*
package com.sachtech.datingapp.ui.explore.fragment

import android.content.Context
import android.os.Bundle
import android.view.View
import androidx.fragment.app.Fragment
import com.cometchat.pro.constants.CometChatConstants
import com.google.android.material.tabs.TabLayout
import com.sachtech.datingapp.R
import com.sachtech.datingapp.cometChat.CometChatMessaging
import com.sachtech.datingapp.data.User
import com.sachtech.datingapp.networking.FirebaseHelper
import com.sachtech.datingapp.ui.explore.adapter.CallTypeListener
import com.sachtech.datingapp.ui.explore.adapter.SearchAdapter
import com.sachtech.datingapp.ui.explore.adapter.listener.CallTypeListener
import com.sachtech.datingapp.ui.home.OnFragmentChange
import gurtek.mrgurtekbase.KotlinBaseFragment
import kotlinx.android.synthetic.main.fragment_search.*


class ExploreFragment : KotlinBaseFragment(R.layout.fragment_search),
    CallTypeListener {

    override fun onAudioCallSelected(uid: String) {
        cometMessage.initiateCall(uid, CometChatConstants.CALL_TYPE_AUDIO)
    }

    override fun onVideoCallSelected(uid: String) {
        cometMessage.initiateCall(uid, CometChatConstants.CALL_TYPE_VIDEO)
    }

    val cometMessage by lazy { CometChatMessaging() }
    var searchAdapter: SearchAdapter? = null
    var arrayList: ArrayList<User>? = null
    val firebaseHelper by lazy { FirebaseHelper() }
    var onFragmentChange: OnFragmentChange?=null

    override fun onAttach(context: Context) {
        super.onAttach(context)
        onFragmentChange?.selectedPos(2)
    }
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        onFragmentChange?.selectedPos(2)
        arrayList = ArrayList()
        setOnClickListener()
        openFragment(LikesFragment())
        searchAdapter =
            SearchAdapter(
                arrayList ,
                this
            )
    }


    private fun setOnClickListener() {
        tabLayout.setOnTabSelectedListener(object :
            TabLayout.BaseOnTabSelectedListener<TabLayout.Tab?> {
            override fun onTabReselected(p0: TabLayout.Tab?) {

            }

            override fun onTabUnselected(p0: TabLayout.Tab?) {

            }

            override fun onTabSelected(p0: TabLayout.Tab?) {
                when (p0!!.position) {
                    0 -> openFragment(LikesFragment())
                    1 -> openFragment(VisitorsFragment())
                    2 -> openFragment(LikedByMeFragment())
                    3 -> openFragment(NotInterestedUserFragment())
                    4 -> openFragment(BlockUserFragment())
                }
            }
        })

    }

    private fun openFragment(kClass: Fragment) {
        val videoFragment = kClass
        val transaction = childFragmentManager.beginTransaction()
        transaction.replace(R.id.exploreView, videoFragment).commit()
        // baselistener.addChildFragment(childFragmentManager, R.id.exploreView, kClass)
    }

}
*/
